### PR TITLE
SALTO-5086: remove old state migration code

### PR DESCRIPTION
**Pull request by ori-moisis**
Cleaning up the code before working on new features in the area

This also contains a couple of unit test fixes in other packages

---

_Additional context for reviewer_

---
_Release Notes_: 
Core:
- Removed code that would migrate very old state files to the "new" format - this would only affect workspaces that were not used for over 10 months

---
_User Notifications_: 
_None_

### CODE REVIEW

File: packages/cli/test/command_builder.test.ts [hunk 1]
```diff
@@ -117,23 +117,38 @@ describe('Command builder', () => {
...
-          config,
-          workspacePath: '.',
-          telemetry,
-          spinnerCreator,
-          commanderInput: [
-            'str',
-            ['str2', 'str3'],
-            { bool: true, choices: 'a', verbose: true },
-          ],
+      describe('when log level is above debug', () => {
```
Comment (ori-moisis) [inline packages/cli/test/command_builder.test.ts:120]: unrelated - this was just annoying since it would fail if the test was run from a terminal where `SALTO_LOG_LEVEL=debug` was defined

File: packages/workspace/test/workspace/workspace.test.ts [hunk 2]
```diff
@@ -2007,13 +2007,18 @@ describe('workspace', () => {
...
     const durationAfterLastModificationMs = 1000 * 60 * durationAfterLastModificationMinutes
     beforeEach(async () => {
       now = Date.now()
-      jest.spyOn(Date, 'now').mockImplementation(() => now)
+      mockDateNow = jest.spyOn(Date, 'now')
+      mockDateNow.mockImplementation(() => now)
       modificationDate = new Date(now - durationAfterLastModificationMs)
     })
+    afterEach(() => {
+      mockDateNow.mockRestore()
```
Comment (ori-moisis) [inline packages/workspace/test/workspace/workspace.test.ts:2020]: unrelated fix - this would mock `Date.now` and not restore it, causing other tests to fail randomly

File: packages/workspace/test/workspace/workspace.test.ts [hunk 3]
```diff
@@ -2007,13 +2007,18 @@ describe('workspace', () => {
   describe('getStateRecency', () => {
     let now: number
     let modificationDate: Date
+    let mockDateNow: jest.SpiedFunction<typeof Date.now>
     const durationAfterLastModificationMinutes = 7
     const durationAfterLastModificationMs = 1000 * 60 * durationAfterLastModificationMinutes
     beforeEach(async () => {
       now = Date.now()
-      jest.spyOn(Date, 'now').mockImplementation(() => now)
+      mockDateNow = jest.spyOn(Date, 'now')
```
Comment (ori-moisis) [inline packages/workspace/test/workspace/workspace.test.ts:2015]: unrelated fix - this would mock `Date.now` and not restore it, causing other tests to fail randomly

File: packages/workspace/test/workspace/workspace.test.ts [hunk 4]
```diff
@@ -2964,10 +2969,13 @@ describe('workspace', () => {
   describe('accountConfig', () => {
     let workspace: Workspace
     let adaptersConfig: MockInterface<AdaptersConfigSource>
+    let resolveMock: jest.SpiedFunction<typeof resolve>
     const configElemId = new ElemID('dummy', 'new')
     const configObjectType = new ObjectType({ elemID: configElemId })
     const configInstanceElement = new InstanceElement('aaa', configObjectType)
-    const resolveMock = jest.spyOn(expressionsModule, 'resolve')
```
Comment (ori-moisis) [inline packages/workspace/test/workspace/workspace.test.ts:2970]: unrelated fix - this would mock the `resolve` function even if this test suite wasn't supposed to run.
this caused other tests to fail if they ran before this suite, or if this suite didn't run (e.g - when debugging a single test in this file)

File: packages/file/src/gzip.ts [hunk 5]
```diff
@@ -37,30 +31,3 @@ export const createGZipWriteStream = (
...
   createGzip(),
 ])
-
-
-//
-// the rest of this file contains backward-compatibility with old state file -
-// should be removed once all workspaces have been upgraded (see SALTO-3149)
-//
-
-export const isOldFormatStateZipFile = async (zipFilename: string): Promise<boolean> => {
```
Comment (yuribro) [inline packages/file/src/gzip.ts:47]: Since these are exported from the package, do we need to do anything special? @deprecate them first? or add it to the release notes?

**Review by yuribro — COMMENTED** [review]
looks good to me, are we waiting for @netama to approve?

File: packages/file/src/gzip.ts (hunk 5 above)
Comment (ori-moisis) [reply packages/file/src/gzip.ts:47]: well, technically all of our packages are version 0 (which means API breakage can happen).
I think we should be fine with this kind of breakage of these specific functions, I only worry about breakage with functions and APIs that could conceivably be used by external code, I don't think these qualify

File: packages/core/src/local-workspace/state.ts [hunk 6]
```diff
@@ -193,18 +160,9 @@ export const localState = (
...
-      return currentFilePaths
-    }
-    const oldStateFilePath = `${filePrefix}${ZIPPED_STATE_EXTENSION}`
-    if (await exists(oldStateFilePath)) {
-      pathToClean = oldStateFilePath
-      return [oldStateFilePath]
-    }
-    return []
-  }
+  const getRelevantStateFiles = async (): Promise<string[]> => (
```
[the remark this answers is not in the document]
Comment (ori-moisis) [reply packages/core/src/local-workspace/state.ts]: probably not

**Review by netama — APPROVED** [review]
